### PR TITLE
EVG-13855 Create TupleSelect Component

### PR DESCRIPTION
**Pull request by khelif96**

### CODE REVIEW

File: src/components/TupleSelect/TupleSelect.test.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    key: "Enter",
+    keyCode: 13,
+  });
+  const { location } = history;
+  expect(location.search).toBe(
+    `?${ProjectFilterOptions.BuildVariant}=some-filter`
+  );
+});
+
+// test("Should allow multiple input filters for different keys as query params", () => {
```
Comment (khelif96) [inline src/components/TupleSelect/TupleSelect.test.tsx:142]: I'm still trying to write this test but theres an issue with the antd select component and react-testing-library. 
https://stackoverflow.com/questions/61080116/ant-design-v4-breaks-react-testing-library-tests-for-select-and-autocomplete

File: src/components/TupleSelect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      </Select>
+      <Input
+        aria-label="Select Text Input"
+        data-cy="tuple-select-input"
+        value={input}
+        onChange={(e) => setInput(e.target.value)}
+        style={{ width: "70%" }}
+        placeholder={selectedOption.placeHolderText}
+        suffix={
+          <Icon
```
Comment (SupaJoon) [inline src/components/TupleSelect/index.tsx:54]: can we give this glyph a pointer: 'cursor'

File: src/components/TupleSelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    updateQueryParams({ [selected]: updatedParams });
+    setInput("");
+  };
+  const selectedOption = options.find((o) => o.value === selected);
+  return (
+    <Input.Group compact>
+      <Select
+        style={{ width: "30%" }}
+        value={selected}
+        onChange={(v) => setSelected(v)}
```
Comment (SupaJoon) [inline src/components/TupleSelect/index.tsx:36]: I think this can be `setSelected`

File: src/components/TupleSelect/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      />
+    </Input.Group>
+  );
+};
+
+const upsertQueryParam = (params: string[] | string, value: string) => {
+  if (params === undefined) {
+    return [value];
+  }
+  if (typeof params === "string" && params === value) {
```
Comment (SupaJoon) [inline src/components/TupleSelect/index.tsx:71]: this function sometimes returns a string and sometimes returns an array. is that intentional?

File: src/components/TupleSelect/index.tsx (hunk 3 above)
Comment (khelif96) [reply src/components/TupleSelect/index.tsx]: I tried this but ran into some type errors with the antd select component exporting a number type while setSelected expects a string

File: src/components/TupleSelect/index.tsx (hunk 4 above)
Comment (khelif96) [reply src/components/TupleSelect/index.tsx]: Updated so its more consistent.

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**